install/Knative-with-ICP.md

```diff
@@ -0,0 +1,185 @@
+
+2. [Set up the installation environment](https://www.ibm.com/support/knowledgecenter/SSBS6K_3.1.1/installing/install_containers.html#setup)
+
+3. [Customize your cluster](https://www.ibm.com/support/knowledgecenter/SSBS6K_3.1.1/installing/install_containers.html#customize)
+
+4. [Set up Docker for your cluster nodes](https://www.ibm.com/support/knowledgecenter/SSBS6K_3.1.1/installing/install_containers.html#cluster_nodes)
+
+5. [Deploy the environment](https://www.ibm.com/support/knowledgecenter/SSBS6K_3.1.1/installing/install_containers.html#deploy)
+
+6. [Verify the status of your installation](https://www.ibm.com/support/knowledgecenter/SSBS6K_3.1.1/installing/install_containers.html#verify)
+
+## Installing Istio
+
+Follow the [instructions](https://istio.io/docs/setup/kubernetes/quick-start-ibm/#ibm-cloud-private) to install and run Istio in [IBM Cloud Private](https://www.ibm.com/cloud/private).
+
+## Installing Knative components
+
+You can install the Knative Serving and Build components together, or Build on its own.
+
+### Installing Knative Serving and Build components
+
+Run the following command to deploy [Knative Serving](https://github.com/knative/serving) and [Knative Build](https://github.com/knative/build)
+
+```
+curl -L https://github.com/knative/serving/releases/download/v0.2.2/release-lite.yaml \
+  | sed 's/LoadBalancer/NodePort/' \
+  | kubectl apply --filename -
+```
+
+### Installing Knative Build only
+
+Just replace `release-lite.yaml` to `build.yaml` file, the other steps are all the same as above.
+
+```
+curl -L https://github.com/knative/serving/releases/download/v0.2.2/build.yaml \
+  | sed 's/LoadBalancer/NodePort/' \
+  | kubectl apply --filename -
+```
+
+### Update the image security policy
+You need to update the [image security policy](https://www.ibm.com/support/knowledgecenter/SSBS6K_3.1.1/manage_images/image_security.html) to allow the cluster to pull the Knative image when you enable the `image-security-enforcement` at the time of [IBM Cloud Private](https://www.ibm.com/cloud/private) installation.
+
+Enter the following command to view the current image security policy:
+```
+kubectl get clusterimagepolicies
+
+NAME                                    AGE
+ibmcloud-default-cluster-image-policy   27m
+```
+
+Then edit the image security policy.
+```
+kubectl edit clusterimagepolicies ibmcloud-default-cluster-image-policy
+```
+
+Update `spec.repositories` by adding `"gcr.io/knative-releases/*"`
+```yaml
+spec:
+  repositories:
+  - name: "gcr.io/knative-releases/*"
+```
+
+### Update pod security policy
+Configure the namespaces `knative-serving`, `knative-build`, `knative-monitoring` and `knative-eventing` into pod security policy `ibm-privileged-psp`. The step as follows:
+
+1. Check the pod security policy in [IBM Cloud Private](https://www.ibm.com/cloud/private).
+```
+kubectl get psp
```

Comment:
Global comment: 
Please indent all these code blocks to nest them under the steps
```suggestion
   kubectl get psp
```